src/interface.rs

```diff
@@ -400,6 +338,216 @@ impl<'a> Interface<'a> {
+                return true;
+            }
+            Key::Ctrl('c')
+            | Key::Ctrl('d')
+            | Key::Ctrl('g')
+            | Key::Ctrl('z')
+            | Key::Esc
+            | Key::Ctrl('r') => {
+                self.run = false;
+                self.input.clear();
+                return true;
+            }
+            Key::Ctrl('b') => self.input.move_cursor(Move::Backward),
+            Key::Ctrl('f') => self.input.move_cursor(Move::Forward),
+            Key::Ctrl('a') => self.input.move_cursor(Move::BOL),
+            Key::Ctrl('e') => self.input.move_cursor(Move::EOL),
+            Key::Ctrl('w') | Key::Alt('\x08') | Key::Alt('\x7f') => {
+                self.input.delete(Move::BackwardWord);
+                self.refresh_matches();
+            }
+            Key::Alt('d') => {
+                self.input.delete(Move::ForwardWord);
+                self.refresh_matches();
+            }
+            Key::Ctrl('v') => {
+                self.debug = !self.debug;
+            }
+            Key::Alt('b') => self.input.move_cursor(Move::BackwardWord),
+            Key::Alt('f') => self.input.move_cursor(Move::ForwardWord),
+            Key::Left => self.input.move_cursor(Move::Backward),
+            Key::Right => self.input.move_cursor(Move::Forward),
+            Key::Up | Key::PageUp | Key::Ctrl('p') => self.move_selection(MoveSelection::Up),
+            Key::Down | Key::PageDown | Key::Ctrl('n') => self.move_selection(MoveSelection::Down),
+            Key::Ctrl('k') => {
+                self.input.delete(Move::EOL);
+                self.refresh_matches();
+            }
+            Key::Ctrl('u') => {
+                self.input.delete(Move::BOL);
+                self.refresh_matches();
+            }
+            Key::Backspace | Key::Ctrl('h') => {
+                self.input.delete(Move::Backward);
+                self.refresh_matches();
+            }
+            Key::Delete => {
+                self.input.delete(Move::Forward);
+                self.refresh_matches();
+            }
+            Key::Home => self.input.move_cursor(Move::BOL),
+            Key::End => self.input.move_cursor(Move::EOL),
+            Key::Char(c) => {
+                self.input.insert(c);
+                self.refresh_matches();
+            }
+            Key::F(2) => {
+                if !self.matches.is_empty() {
+                    self.menu_mode = MenuMode::ConfirmDelete;
+                }
+            }
+            Key::Ctrl(_c) => {
+                //                      self.debug(&mut screen, format!("Ctrl({})", c))
+            }
+            Key::Alt(_c) => {
+                //                      self.debug(&mut screen, format!("Alt({})", c))
+            }
+            Key::F(_c) => {
+                //                      self.debug(&mut screen, format!("F({})", c))
+            }
+            Key::Insert | Key::Null | Key::__IsNotComplete => {}
+            _ => {}
```

Comment:
Does it error if you don't have this line? I thought `Key::Char(c)`, `Key::Ctrl(_c)`, `Key::Alt(_c)`, `Key::F(_c)` covered the match.